tooling/minimize/tests/pass/slice.rs

```diff
@@ -0,0 +1,27 @@
+extern crate intrinsics;
+use intrinsics::*;
+
+#[repr(C)]
+struct SlicePointerTuple<T> {
+    begin: *const T,
+    len: usize,
+}
+
+pub fn print_some_elements(a: &[i32]) {
+    print(a[1]);
+    print(a[2]);
+    print(a[3]);
+}
+
+fn main() {
+    let x: &[i32; 5] = &[50, -40, 30, -20, 10];
+
+    // We do not have normal unsizing yet, so we cheat.
+    let ptr = SlicePointerTuple { begin: x[0] as *const i32, len: 5 };
```

Comment:
Here you are casting `50` to a pointer, is that your intent? I think you mean `&x[0] as *const i32`, which is better written as `x.as_ptr()`.

That's where the `predict` call came from.

Reply:
Was not my intent, but it did come up during debugging, but I wrongly assumed you cannot cast i32 to a pointer (only usize). But yeah, did was indeed the problem (altho I cannot use `as_ptr` since it does an implicit unsizing).

Reply:
> I wrongly assumed you cannot cast i32 to a pointer (only usize)

That is a common footgun, indeed. There's a long-standing open issue somewhere to warn against non-usize-int-to-ptr casts, but so far it hasn't been implemented yet.